File: server/DTOs/CategoryDtos.cs
namespace FeelJournal.DTOs;

public class CategoryDto
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
}

public class CreateCategoryRequest
{
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = "NEUTRAL";
}

public class UpdateCategoryRequest
{
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = "NEUTRAL";
}

File: app/src/main/java/com/example/feel_journal/data/remote/dto/CategoryDtos.kt
package com.example.feel_journal.data.remote.dto

import com.google.gson.annotations.SerializedName

data class CreateCategoryRequest(
    @SerializedName("name") val name: String,
    @SerializedName("type") val type: String = "NEUTRAL"
)

data class UpdateCategoryRequest(
    @SerializedName("name") val name: String,
    @SerializedName("type") val type: String = "NEUTRAL"
)
